### DISCUSSION TITLE
Are there any good knowledge sources on best practices in actor-based architecture?

### DISCUSSION THREAD

**Discussion by svetlyak40wt** [mdbergmann/cl-gserver#89] [General]
I really thinking about using actors in two areas:

* a web scraper [ScrapyCL](https://40ants.com/scrapycl/)
* a Telegram bot framework [cl-telegram-bot](https://40ants.com/cl-telegram-bot/)

Tried to search something about Actors on YouTube, but only found videos about basic conceptions. What I'm looking for is a more depth and practical articles or books about actors, usage patterns, etc.

Could you advice something?

**Comment by mdbergmann** [comment]
Well, hmm. I don't know any book about actors as such.
Initially, the first steps I tried to model Sento after Erlang. Erlang is probably THE prominent actor based runtime. So the Actor-Cell and its 'cast' and 'call' stuff is still coming from there. But the threading model in Common Lisp is much different than Erlangs, so as a next version I've modelled it much more for how Akka (https://akka.io) work. This is what I'm working with at work. JVM also has heavy-weight OS threads only (although in Java 21 now are also virtual threads).

TL;DR:
Maybe it would be worthwhile to look at available books about Akka? Just to get the concepts, dos and don'ts and so on. But Akka comes with a ton of more features that are not in Sento, I.e. remoting, clustering, etc. But the foundations are similar.

I'm not what you read about the basic concepts. Are they clear?

**Reply by svetlyak40wt** [reply]
Thank you for the idea, I'll start from the [Shopping Cart](https://doc.akka.io/guide/microservices-tutorial/overview.html) Akka tutorial  and will try to reproduce it with Sento!
